### PR TITLE
Add the feature to delete a job post - front end

### LINKED ISSUE #30: Add feature to delete a job post
As per title
Change the job post status to DELETED instead of truly deleting in the database
So far backend section finishes. Front end section has not started yet.

### PR DESCRIPTION
**Pull request by SibingWu**
Fixed #30 

* Front end section

### CODE REVIEW

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * This file is specific to index.html for job-details. 
+ * It renders the fields on the page dynamically,
+ * and send PATCH request to mark the current job post as deleted.
+ */
+
+/**
+ * Import statements are static so its parameters cannot be dynamic.
+ * TODO(issue/22): figure out how to use dynamic imports
+ */
+import {AppStrings} from "../strings.en.js";
```
Comment (riya-n) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: nit: single quotes (also for below)

File: walk-in-interview/src/main/webapp/job-details/index.html [hunk 2]
```diff
@@ -3,12 +3,18 @@
 
 <head>
     <meta charset="UTF-8">
-    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="../style.css">
 </head>
 
+<script type="module" src="../delete-job/script.js"></script>
```
Comment (riya-n) [inline walk-in-interview/src/main/webapp/job-details/index.html]: I think you can add this below the <body> portion of the HTML

File: walk-in-interview/src/main/webapp/job-details/index.html [hunk 3]
```diff
@@ -3,12 +3,18 @@
...
+
 <body>
     <div class="header-container">
         <button onclick="location.href='../index.html'">Back to Homepage</button>
```
Comment (riya-n) [inline walk-in-interview/src/main/webapp/job-details/index.html:12]: maybe add a TODO to render this page dynamically

**Review by riya-n — APPROVED** [review]

File: walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java [hunk 4]
```diff
@@ -51,4 +54,18 @@ public void doPatch(HttpServletRequest request, HttpServletResponse response) {
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
         }
     }
+
+    /** Gets jobId received from client. */
+    private String getJobId(HttpServletRequest request) throws IOException, IllegalArgumentException {
+        // Parses job object from the POST request
```
Comment (htthach) [inline walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java]: I think this comment is not correct here?

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ * Import statements are static so its parameters cannot be dynamic.
+ * TODO(issue/22): figure out how to use dynamic imports
+ */
+import {AppStrings} from '../strings.en.js';
+
+import {setErrorMessage} from '../common-functions.js';
+
+const STRINGS = AppStrings['delete-job'];
+const HOMEPAGE_PATH = '../index.html';
+const RESPONSE_ERROR = 'There was an error while deleting the job post. Please try again';
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: Nit: look like this is ERROR_MESSAGE?

Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: Is this string part of string set to be translated?

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  .catch((error) => {
+    setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ RESPONSE_ERROR,
+      /* includesDefault= */false);
+    console.log('error', error);
+  });
+}
+
+const deleteButtonElement = document.getElementById('delete');
+deleteButtonElement.addEventListener('click', () => {
+const jobId = getJobId();
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: Nit: Is indentation missing for this line?

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import {AppStrings} from '../strings.en.js';
+
+import {setErrorMessage} from '../common-functions.js';
+
+const STRINGS = AppStrings['delete-job'];
+const HOMEPAGE_PATH = '../index.html';
+const RESPONSE_ERROR = 'There was an error while deleting the job post. Please try again';
+
+window.onload = () => {
+  renderDeleteButton();
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js:19]: Just curious, if I revisit this page after deleted, will delete button show again?

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      /* includesDefault= */false);
+    console.log('error', error);
+  });
+}
+
+const deleteButtonElement = document.getElementById('delete');
+deleteButtonElement.addEventListener('click', () => {
+const jobId = getJobId();
+  if (jobId === '') {
+    setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ 'Empty Job Id found.' + RESPONSE_ERROR,
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: Do we still need to append this generic error message at the end?

File: walk-in-interview/src/main/webapp/delete-job/script.js (hunk 7 above)
Comment (SibingWu) [reply walk-in-interview/src/main/webapp/delete-job/script.js:19]: After deleted, there is no way to access this page with specific jobId

File: walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java [hunk 9]
```diff
@@ -2,19 +2,25 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
 
 /** Servlet that handles changing status of the existing job posts to DELETED. */
 @WebServlet("/jobs/delete")
 public final class MarkJobDeleteServlet extends HttpServlet {
     private static final String PATCH_METHOD_TYPE = "PATCH";
-    private static final String JOB_ID_FIELD = "jobId";
+    private static final long TIMEOUT = 5;
```
Comment (htthach) [inline walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java]: Nit: is it in seconds? Maybe name it TIMEOUT_SECONDS?

File: walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java [hunk 10]
```diff
@@ -37,18 +43,33 @@ public void service(HttpServletRequest request, HttpServletResponse response) th
...
             // TODO(issue/47): use custom exceptions
             System.err.println("Error occur: " + e.getCause());
             // Sends the fail status code in the response
             response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
         }
     }
+
+    /** Gets jobId received from client. */
+    private String getJobId(HttpServletRequest request) throws IOException, IllegalArgumentException {
+        // Parses job Id from the POST request
```
Comment (htthach) [inline walk-in-interview/src/main/java/com/google/job/servlets/MarkJobDeleteServlet.java]: I think this commend is not what the code is doing?

File: walk-in-interview/src/main/webapp/delete-job/script.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+      /* includesDefault= */false);
+    console.log('error', error);
+  });
+}
+
+const deleteButtonElement = document.getElementById('delete');
+deleteButtonElement.addEventListener('click', () => {
+  const jobId = getJobId();
+  if (jobId === '') {
+    setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */ 'Empty Job Id found.',
```
Comment (htthach) [inline walk-in-interview/src/main/webapp/delete-job/script.js]: Should this message be part of the translation?

**Review by htthach — APPROVED** [review]

**Pull request merged**